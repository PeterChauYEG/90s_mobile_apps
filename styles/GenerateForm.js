// react native
import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  body: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5
  },
  textInput: {
    height: 40,
    width: 100
  },
})

export default styles