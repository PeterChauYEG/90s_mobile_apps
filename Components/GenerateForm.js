/**
 *
 * @format
 * @flow
 */

// react
import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'

// react native
import { Button, Slider, Text, TextInput, View } from 'react-native'

// watch
import * as watch from 'react-native-watch-connectivity'

// styles
import styles from './styles/GenerateForm'

type Props = {
  lyricGenerationRequest: (number, string) => void
}
type State = {
  nChars: number,
  sample: string
}
class GenerateForm extends Component<Props, State> {
  static propTypes = {
    lyricGenerationRequest: PropTypes.func.isRequired
  }
  
  static defaultProps = {}
  
  constructor (props: Props) {
    super(props)

    this.state = {
      nChars: 50,
      sample: 'sweet dreams are made of '
    }
    
  }

  onNCharsChange = (number: number) => {
    this.setState({ nChars: number })
  }

  onSampleChange = (text: string) => {
    this.setState({ sample: text })
  }

  render () {
    const { lyricGenerationRequest } = this.props
    const { nChars, sample } = this.state

    // creates a subscription for messages from the watch 
    // then makes a generate request
    watch.subscribeToMessages((err, message, reply) => {
      if (err) {
        console.log(err)
        return
      }
    
      reply({text: "message received!"})
      if (message.action === "generate") {
        this.setState({ sample: message.sample })
        lyricGenerationRequest(nChars, message.sample)
      }
        
    })
    
    return (
      <Fragment>
        <Text style={styles.body}>Enter some initial lyrics</Text>
        <TextInput
          onChangeText={this.onSampleChange}
          placeholder='sweet dreams are made of these'
          style={styles.textInput}
          value={sample}
        />
        <Text style={styles.body}>Enter of characters to generate</Text>
        <Slider
          maximumValue={500}
          minimumValue={10}
          onValueChange={this.onNCharsChange}
          step={10}
          style={styles.textInput}
          value={nChars}
        />
        <Text>{nChars}</Text>
        <Button
          accessibilityLabel='Generate lyrics from your initial lyrics'
          color='#841584'
          onPress={() => lyricGenerationRequest(nChars, sample)}
          title='Generate'
        />
      </Fragment>
    )
  }
}

export default GenerateForm
