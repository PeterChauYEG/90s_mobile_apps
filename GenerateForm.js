/**
 *
 * @format
 * @flow
 */

// react
import React, { Component, Fragment } from 'react'

// react native
import { Button, Slider, Text, TextInput, View } from 'react-native'

// styles 
import styles from './styles/GenerateForm'

type Props = {}
class GenerateForm extends Component<Props> {
  constructor (props) {
    super(props)

    this.state = {
      nChars: 50,
      sample: 'sweet dreams are made of '
    }
  }

  onNCharsChange = text => {
    this.setState({ nChars: text })
  }

  onSampleChange = text => {
    this.setState({ sample: text })
  }

  render () {    
    const { lyricGenerationRequest } = this.props
    const { nChars, sample } = this.state
    
    return (
      <Fragment>
        <Text style={styles.body}>Enter some initial lyrics</Text>
        <TextInput
          onChangeText={this.onSampleChange}
          placeholder='sweet dreams are made of these'
          style={styles.textInput}
          value={sample}
        />
        <Text style={styles.body}>Enter of characters to generate</Text>
        <Slider
          maximumValue={500}
          minimumValue={10}
          onValueChange={this.onNCharsChange}
          step={10}
          style={styles.textInput}
          value={nChars}
        />
        <Text>{nChars}</Text>
        <Button
          accessibilityLabel='Generate lyrics from your initial lyrics'
          color='#841584'
          onPress={() => lyricGenerationRequest(nChars, sample)}
          title='Generate'
        />            
      </Fragment>
    )
  }
}

export default GenerateForm
