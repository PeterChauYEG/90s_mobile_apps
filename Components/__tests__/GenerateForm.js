import React from 'react'
import GenerateForm from '../GenerateForm'
import renderer from 'react-test-renderer'

describe('ErrorOccurred', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<GenerateForm lyricGenerationRequest={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  // TODO: - write test for button click
  // TODO: - write test for slider change
  // TODO: - write test for input change
})
