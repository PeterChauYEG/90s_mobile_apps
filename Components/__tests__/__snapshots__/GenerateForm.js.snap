// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorOccurred renders correctly 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Enter some initial lyrics
  </Text>,
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="sweet dreams are made of these"
    style={
      Object {
        "height": 40,
        "width": 100,
      }
    }
    underlineColorAndroid="transparent"
    value="sweet dreams are made of "
  />,
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Enter of characters to generate
  </Text>,
  <RCTSlider
    disabled={false}
    enabled={true}
    maximumValue={500}
    minimumValue={10}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onValueChange={[Function]}
    step={10}
    style={
      Array [
        Object {
          "height": 40,
        },
        Object {
          "height": 40,
          "width": 100,
        },
      ]
    }
    value={50}
  />,
  <Text>
    50
  </Text>,
  <View
    accessibilityLabel="Generate lyrics from your initial lyrics"
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Generate
      </Text>
    </View>
  </View>,
]
`;
